src/sentry/static/sentry/app/views/settings/projectSourceMaps/detail/index.tsx

```diff
@@ -0,0 +1,210 @@
+  addSuccessMessage,
+} from 'app/actionCreators/indicator';
+
+import SourceMapsFileRow from './sourceMapsFileRow';
+
+type RouteParams = {orgId: string; projectId: string; version: string};
+
+type Props = RouteComponentProps<RouteParams, {}> & {
+  organization: Organization;
+  project: Project;
+};
+
+type State = AsyncView['state'] & {
+  files: SourceMap[];
+};
+
+class ProjectSourceMaps extends AsyncView<Props, State> {
+  getTitle() {
+    const {projectId, version} = this.props.params;
+
+    return routeTitleGen(t('Source Maps %s', formatVersion(version)), projectId, false);
+  }
+
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      files: [],
+    };
+  }
+
+  getEndpoints() {
+    const {location} = this.props;
+
+    const endpoints: ReturnType<AsyncView['getEndpoints']> = [
+      ['files', this.getFilesUrl(), {query: {query: location.query.query}}],
+    ];
+
+    return endpoints;
+  }
+
+  getFilesUrl() {
+    const {orgId, projectId, version} = this.props.params;
+
+    return `/projects/${orgId}/${projectId}/releases/${encodeURIComponent(
+      version
+    )}/files/`;
+  }
+
+  handleSearch = (query: string) => {
+    const {location, router} = this.props;
+
+    router.push({
+      ...location,
+      query: {...location.query, cursor: undefined, query},
+    });
+  };
+
+  handleDelete = async (id: string) => {
+    addLoadingMessage(t('Removing file...'));
+
+    try {
+      await this.api.requestPromise(`${this.getFilesUrl()}${id}/`, {
+        method: 'DELETE',
+      });
+      // We might want to refetch the files here to not mess up pagination
+      this.setState(state => ({files: state.files.filter(f => f.id !== id)}));
+      addSuccessMessage(t('File removed.'));
+    } catch {
+      addErrorMessage(t('Unable to remove file. Please try again.'));
+    }
+  };
+
+  getQuery() {
+    const {query} = this.props.location.query;
+
+    return typeof query === 'string' ? query : undefined;
+  }
+
+  getEmptyMessage() {
+    if (this.getQuery()) {
+      return t('There are no source maps that match your search.');
+    }
+
+    return t('There are no source maps for this release.');
+  }
+
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderFiles() {
+    const {files} = this.state;
+    const fileApiUrl = this.api.baseUrl + this.getFilesUrl();
+
+    if (!files.length) {
+      return null;
+    }
+
+    return files.map(file => {
+      return (
+        <SourceMapsFileRow
+          key={file.id}
+          file={file}
+          onDelete={this.handleDelete}
+          downloadUrl={`${fileApiUrl}${file.id}/?download=1`}
+        />
+      );
+    });
+  }
+
+  renderBody() {
+    const {loading, files, filesPageLinks} = this.state;
+    const {version, orgId, projectId} = this.props.params;
+    const {project} = this.props;
+
+    return (
+      <React.Fragment>
+        <SettingsPageHeader
+          title={`${t('Source Maps in')} ${formatVersion(version)}`}
```

Comment:
You should use a %s placeholder or tct here. Concatenating strings makes translations hard.